test/qos2.js

```diff
@@ -129,9 +129,10 @@ test('call published method with client with QoS 2', function (t) {
   }
 
   broker.published = function (packet, client, cb) {
-    t.ok(client, 'client must be passed to published method')
-
-    cb()
+    if (packet.topic.split('/').pop() !== 'clientDisconnect') {
```

Comment:
I think its already mentioned in the published method block. I'll add it here though.